ckanext/data_qld/controller_functions.py

```diff
@@ -57,22 +61,61 @@ def open_datarequest(id):
         return abort(404, _('Data Request %s not found') % id)
     except NotAuthorized as e:
         log.warn(e)
-        return abort(403, _('You are not authorized to open the Data Request %s' % id))
+        return abort(
+            403, _('You are not authorized to open the Data Request %s' % id))
 
 
-def show_schema(dataset_id, resource_id):
-    data_dict = {'id': resource_id}
+def show_resource_schema(dataset_id, resource_id):
     context = _get_context()
 
+    data = _get_resource_data(resource_id, context)
+    schema_data = data.get('schema')
+    c.schema_data = json.dumps(schema_data, indent=2, sort_keys=True)
```

Comment:
This is not the recommended way to pass data to a template. It is better to use the `extra_vars` parameter of the `render` function.

Reply:
Just copied the previous approach. Will be changed with other stuff.